hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -258,6 +259,13 @@ void tryStartJob(Supplier<Long> executionIdSupplier) {
             if (scheduleRestartIfQuorumAbsent() || scheduleRestartIfClusterIsNotSafe()) {
                 return null;
             }
+            Version jobClusterVersion = mc.jobRecord().getClusterVersion();
+            Version currentClusterVersion = mc.nodeEngine().getClusterService().getClusterVersion();
+            if (!jobClusterVersion.equals(currentClusterVersion)) {
```

Comment:
Isn't `JobConfig.clusterVersion` nullable? The nightly test below actually specifies `null` in one of the cases it seems.